lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -167,6 +209,57 @@ void should_OnlyReturnTasksFromCorrectWorkbaskets_When_UserHasNoPermissionToOneW
           .contains(taskSummary1, taskSummary2)
           .doesNotContain(taskSummary3, taskSummary4, taskSummary5);
     }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnEmptyList_When_WorkbasketOfTaskHasNoReadTasksPerm() {
+      List<TaskSummary> list = taskService.createTaskQuery().idIn(taskSummary3.getId()).list();
+      assertThat(list.isEmpty());
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ThrowException_When_QueryByWorkbasketThatHasOpenReadButNoReadTasksPermission() {
+      assertThatThrownBy(
+              () ->
+                  taskService
+                      .createTaskQuery()
+                      .workbasketIdIn(wbWithoutReadTasksPerm.getId())
+                      .list())
+          .isInstanceOf(NotAuthorizedToQueryWorkbasketException.class);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnEmptyList_When_WorkbasketOfTaskHasReadTasksButNoReadPerm() {
+      List<TaskSummary> list = taskService.createTaskQuery().idIn(taskSummary7.getId()).list();
+
+      assertThat(list).isEmpty();
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_QueryByTaskId_When_WorkbasketHasReadAndReadTasksButNoOpenPerm() {
```

Comment:
Maybe another test with a query for task7 and task8 makes sense, making sure that only task8 is returned?

Reply:
But Task 8 has no Open Permission, so it should also not be returned